app/src/org/commcare/utils/HereFunctionHandler.java

```diff
@@ -85,20 +90,33 @@ public void refreshLocation() {
     }
 
     @Override
-    public void onLocationChanged(Location location) {
-        this.location = toGeoPointData(location);
-        Log.i("HereFunctionHandler", "location has been set to " + this.location.getDisplayText());
+    public void onLocationChanged(Location updatedLocation) {
+        this.location = updatedLocation;
+        Log.i("HereFunctionHandler", "location has been set to " + this.location);
 
         if (this.location.getAccuracy() <= GeoUtils.ACCEPTABLE_ACCURACY) {
             locationGoodEnough = true;
             forbidGpsUse();
         }
 
-        if (entitySelectActivity != null) {
+
+        if (shouldRefreshEntityList()) {
```

Comment:
I would argue that it makes more sense to write this as `if (entitySelectActivity != null && shouldRefreshEntityList())`, since the null-check isn't part of the logic implied by the method name; feels more like just its own separate concern/requirement. It also allows for short-circuiting in the event that `entitySelectActivity` is null.